src/core/Messages.ts

```diff
@@ -4,11 +4,20 @@
 import {classToPlain, plainToClass} from "class-transformer";
 import "reflect-metadata";
 
-export class ButtplugMessage {
+export abstract class ButtplugMessage {
 
   constructor(public Id: number) {
   }
 
+  abstract get SchemaVersion(): number;
+
+  public DowngradeMessage(): ButtplugMessage {
+    if (this.SchemaVersion === 0) {
+      return new Error("Cannot downgrade a message at spec version 0.", ErrorClass.ERROR_MSG, this.Id);
```

Comment:
In most cases, this is probably OK, but imagine if we added a new server side message that was sent out from some subsystem (unaware of the client version). Returning this error here will actually cause the error to be sent to the client, which I bet won't be expecting it.

This might be OK in JS, because of how the downgrade logic works, but it'll be another thing to remember when implementing new messages.

Reply:
I suppose we can just log here instead of returning to the client?